pkg/controller/controller.go

```diff
@@ -217,10 +217,10 @@ func (ctrl *CSIAttachController) syncVA() {
 }
 
 func (ctrl *CSIAttachController) processFinalizers(pv *v1.PersistentVolume) bool {
-	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == ctrl.attacherName {
-		return true
+	if pv.DeletionTimestamp != nil {
+		return sets.NewString(pv.Finalizers...).Has(GetFinalizerName(ctrl.attacherName))
 	}
-	return false
+	return pv.Spec.CSI != nil && pv.Spec.CSI.Driver == ctrl.attacherName
```

Comment:
I think you can safely remove this `pv.Spec.CSI` check now.

Reply:
Hm, wouldn't the attacher then also consider `PersistentVolume`s whose `.spec.csi.driver` doesn't match to its own attacher name? I guess this was the reason why the check was there in the first place?

Reply:
The finalizer name is derived from CSI driver name, so IMO it's not necessary to check `pv.spec.csi.driver`.

Reply:
OK. I'm not too deep in the details, but got this question: How would a setup with multiple CSI drivers in the same cluster look like? Would I have only one instance of the external-attacher or one per CSI driver?

Reply:
Each driver must have its own external-attacher instance

Reply:
OK, thanks. Suppose the discussed check gets removed - wouldn't these multiple external-attacher instances then all start to reconcile the `PersistentVolume` although at least some of them are not responsible?

Reply:
That should be filtered by finalizer check you added above - each attacher will process only PVs with its own `GetFinalizerName(ctrl.attacherName))`.

Oh, btw, the function should return `false` if deletion timestamp is not set, of course.